### ISSUE TITLE
Qwen3 runtime: degenerate output on raw text generation (independent of tokenizer / chat template)

### ISSUE THREAD

**Issue by michalharakal** [SKaiNET-developers/SKaiNET-transformers#74]
## Problem

The Qwen3 runtime path produces degenerate output **even on raw text generation** with no chat template and no special tokens. This is a runtime-level bug independent of the tokenizer fix that ships in #72 / #73.

## Reproducer

```sh
java --enable-preview --add-modules jdk.incubator.vector -Xmx16g \
  -jar llm-apps/kllama-cli/build/libs/kllama-all.jar \
  -m Qwen3-1.7B-Q8_0.gguf --context=1024 -s 32 -k 0.0 \
  "The capital of France is"
```

**Actual output**:
```
The capital of France is the city of of the the country of of the the country of of the the country of of the the country of of the the country
```

**Expected** (compare to Llama 3.2 1B Q8_0 on the same harness, same prompt, same temperature):
```
The capital of France is Paris. The Eiffel Tower is located in Paris. The Louvre Museum is also located in Paris. The famous French artist Claude Monet was born in
```

The `--demo --template=qwen` path produces similarly degenerate output (` it", in" "  < |user" is?  and6.=' k    22,  12 1 the  the2 la, 2PDF...`) but that's downstream of this same bug — even the no-special-tokens raw path collapses.

## Why this is *not* the tokenizer bug

The tokenizer fix in #72 / commit `8dad771` makes special tokens (`<|im_start|>`, `<|im_end|>`, `<|tool_call>`) atomic. Qwen3 benefits from that on the chat-template path. But the prompt above contains zero special tokens — pure ASCII text. The encoder fragments nothing, the prompt token IDs are correct, the runtime is what produces broken logits.

## Why smoke tests didn't catch it

`tests/smoke/smoke-models.json` lists `Qwen3-1.7B-Q8` with a tool-calling assertion. The bash harness (`tests/smoke/smoke-test.sh:330`) checks for the *literal string* `[Tool Call]` in stdout — which `kllama` only prints if `ToolCallParser` recognises a parseable call. With this output, the parser sees nothing → no `[Tool Call]` → harness records this as `WARN` (no tool call detected) which lives in a different counter than `FAIL`. The summary line still reports overall pass.

## Suspected causes

Qwen3 architectural differences from Llama that the kllama runtime (`LlamaRuntime` reused for Qwen via `LlamaWeightLoader` + `LlamaWeightMapper` in `Main.kt:363`) may not handle correctly:

1. **`q_norm` / `k_norm`** — Qwen3 applies RMSNorm to Q and K projections per-head before RoPE. Llama doesn't have this. If the runtime ignores `attn_q_norm.weight` / `attn_k_norm.weight` GGUF tensors, attention scores will be off-scale.

2. **`rope_theta`** — Qwen3 uses base `1000000` (or `5000000` for some variants); Llama 3.2 uses `500000`. If the runtime is hardcoded or reading the wrong field, RoPE rotations are wrong.

3. **`head_dim`** — Qwen3-1.7B has `head_dim=128` even though `hidden_size / num_attention_heads = 1536/16 = 96`. The runtime may be computing head_dim from hidden/heads instead of reading the explicit `head_dim` field, which would catastrophically misshape Q/K/V.

The repetition collapse pattern (`of of the the country of of`) is consistent with attention not discriminating between positions / KV cache being misindexed — exactly what any of those three would produce.

## Investigation steps (suggested)

- `gguf-utils` / `inspect_model.py` on `Qwen3-1.7B-Q8_0.gguf` to confirm:
  - presence of `attn_q_norm.weight` / `attn_k_norm.weight` per layer
  - `qwen3.rope.freq_base`
  - `qwen3.attention.head_count` vs `qwen3.attention.head_count_kv` vs hidden_size, and whether a `qwen3.attention.key_length` / `head_dim` field is present
- Check `LlamaWeightMapper.map(...)` — does it skip / no-op on `attn_q_norm` / `attn_k_norm`?
- Check `LlamaRuntime.runLayer` — does it apply Q/K norm anywhere?
- Compare with the Gemma 4 path which does have q-norm wired (commits like `c7402e9 fix(gemma4): v_norm + qkNormUnitOffset`), since Gemma 4 hit the same class of bug and we have the diagnostic tooling for it.

## Out of scope

Tool-calling on top of Qwen3 — until raw generation is coherent there is no point validating the chat template path. The smoke matrix's Qwen3 tool-calling claim should probably be downgraded to "loads + decodes without crash" until this is fixed.

## Related

- #72 / #73 — Llama 3 tool calling fix (special-token encoding + Meta-aligned template). Independent of this bug, but the discovery happened together.
- The `feature/llama-tool` branch's transcript shows both bugs side by side: Llama 3.2 1B works end-to-end after the fixes, Qwen3 still produces garbage on raw text.
